libs/core/src/lib/feed-list-item/directives/line-clamp.directive.ts

```diff
@@ -0,0 +1,229 @@
+import { coerceNumberProperty } from '@angular/cdk/coercion';
+import {
+  Directive,
+  ElementRef,
+  EventEmitter,
+  Input,
+  AfterViewInit,
+  OnDestroy,
+  OnChanges,
+  Output,
+  Renderer2
+} from '@angular/core';
+import { fromEvent, Subscription } from 'rxjs';
+import { debounceTime, distinctUntilChanged } from 'rxjs/operators';
+
+@Directive({
+  // tslint:disable-next-line:directive-selector
+  selector: '[fd-lineclamp-target]',
+  exportAs: 'fdLineClampTarget'
+})
+export class LineClampTargetDirective implements OnChanges, AfterViewInit {
+  /**
+   * Target text for clamping
+   */
+  @Input()
+  fdLineClampTargetText: string;
+  /**
+   * Event with target instance for clamping
+   */
+  @Output() update = new EventEmitter<LineClampTargetDirective>();
+  /** @hidden */
+  constructor(private readonly _elementRef: ElementRef) {}
+  /**
+   * Native element of clamping target
+   */
+  get targetElement(): HTMLElement {
+    return this._elementRef.nativeElement;
+  }
+  /** @hidden */
+  ngOnChanges(): void {
+    if (typeof this.targetElement.style.webkitLineClamp === 'undefined') {
+      this.update.emit(this);
+    }
+  }
+  /** @hidden */
+  ngAfterViewInit(): void {
+    this.update.emit(this);
+  }
+}
+
+@Directive({
+  // tslint:disable-next-line:directive-selector
+  selector: '[fd-lineclamp]',
+  exportAs: 'fdLineClamp'
+})
+export class LineClampDirective implements OnChanges, AfterViewInit, OnDestroy {
```

Comment:
Great job with this directive :rocket:, works great on ie11 as well, In feature it can be moved to utils

Reply:
Moved